File: backend/src/main/java/com/okaya/service/ProduitService.java
package com.okaya.service;

import com.okaya.model.Produit;
import com.okaya.repository.ProduitRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.Optional;

/**
 * Service pour la gestion des produits
 */
@Service
public class ProduitService {
    
    @Autowired
    private ProduitRepository produitRepository;
    
    /**
     * Récupérer tous les produits
     */
    public List<Produit> getAllProduits() {
        return produitRepository.findAll();
    }
    
    /**
     * Récupérer un produit par son ID
     */
    public Optional<Produit> getProduitById(Long id) {
        return produitRepository.findById(id);
    }
    
    /**
     * Récupérer un produit par sa référence
     */
    public Optional<Produit> getProduitByReference(String reference) {
        return produitRepository.findByReference(reference);
    }
    
    /**
     * Créer ou mettre à jour un produit
     */
    public Produit saveProduit(Produit produit) {
        return produitRepository.save(produit);
    }
    
    /**
     * Supprimer un produit
     */
    public void deleteProduit(Long id) {
        produitRepository.deleteById(id);
    }
}

File: frontend/src/app/services/produit.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Produit } from '../models/produit.model';

/**
 *  pour la gestion des produits
 */
@Injectable({
  providedIn: 'root'
})
export class ProduitService {
  private apiUrl = 'http://localhost:8080/api/produits';

  constructor(private http: HttpClient) {}

  /**
   * Récupérer tous les produits
   */
  getAll(): Observable<Produit[]> {
    return this.http.get<Produit[]>(this.apiUrl);
  }

  /**
   * Récupérer un produit par ID
   */
  getById(id: number): Observable<Produit> {
    return this.http.get<Produit>(`${this.apiUrl}/${id}`);
  }

  /**
   * Créer un nouveau produit
   */
  create(produit: Produit): Observable<Produit> {
    return this.http.post<Produit>(this.apiUrl, produit);
  }

  /**
   * Mettre à jour un produit
   */
  update(id: number, produit: Produit): Observable<Produit> {
    return this.http.put<Produit>(`${this.apiUrl}/${id}`, produit);
  }

  /**
   * Supprimer un produit
   */
  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
